scripts/featured-svg-to-png.py

```diff
@@ -0,0 +1,109 @@
+import os
+import sys
+import glob
+import yaml
+import pyvips
+
+# This script takes a directory, goes through all blog posts in that directory,
+# and converts any featured images that use SVG to PNG.
+#
+# To use this script:
+#
+# 1. Create an environment and install dependencies:
+#
+#     conda create -n featured-svg-to-png -c conda-forge pyvips pyyaml
+#     conda activate featured-svg-to-png
+#
+# 2. Run the script:
+#
+#     python featured-svg-to-png.py /path/to/mdx-blog-posts
+#
+# 3. Finally, deactivate the environment:
+#
+#     conda deactivate
+
+
+# Function to convert SVG to PNG
+def convert_svg_to_png(svg_path, png_path):
+    try:
+        # 1200 px width comes from the open graph recommendation
+        # See: https://github.com/Quansight/Quansight-website/blob/3d1e583d6f69f12c11438b5cb2f396421fb900e9/libs/shared/ui-components/src/SocialCard/types.ts#L55
+        image = pyvips.Image.thumbnail(svg_path, 1200)
+        image.write_to_file(png_path)
+        print(f"Converted {svg_path} to {png_path}")
+        return True
+    except Exception as e:
+        print(f"Error converting {svg_path} to PNG: {e}")
+        return False
+
+# Function to process markdown files with YAML front matter.
+#
+# It looks for a featured image file pathname in the YAML. If the pathname ends
+# with .svg, it opens the file, converts it to .png, then updates the YAML file
+# pathname to point to the PNG file.
+#
+# Assumes the following relationship between the markdown and image file paths:
+# <markdown_path>/../../public/<svg_path>
+#
+# example.md:
+# ---
+# title: Example Blog Post
+# featuredImage:
+#   src: /posts/example-blog-post/feat.svg
+# ---
+# ...
+def process_markdown_file(file_path):
+    with open(file_path, 'r') as f:
+        file_contents = f.read()
+
+    front_matter, content = file_contents.split('---', 2)[1:]
```

Comment:
Minor thing here, if the `description` field in the header contains `---`, this will split in the wrong place.

A regex-based split here might be better.

But this is for some hypothetical-and-presumably-nonexistent future use of the script (since we're prohibiting SVGs going forward), so I don't think the effort for a change is warranted.

Reply:
I didn't think about that, haha. Good point.

Thanks for reviewing the script! I was hoping another set of eyes would go over it.